### PR TITLE
Update 0.40.0 release notes

### PR DESCRIPTION
**Pull request by alexcrichton**
Not a ton happened in terms of user-facing improvements here so I
outlined some internal changes as well. The cumulative effect of
improving compile times is Sightglass showing 30-40% improvements for
major benchmarks. Additionally I wrote down a note indicating that this
is likely the last `0.*` release and the next release of Wasmtime on
September 20 is planned to be 1.0.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -14,9 +14,60 @@ Unreleased.
...
+
+* Many more instructions are now implemented with ISLE instead of handwritten
+  lowerings.
+* Many improvements to the cranelift-based fuzzing.
+* Many platform improvements for s390x for running `rustc_codegen_cranelift`
+  with features like `i128` support, supporting more ABIs, etc.
+* Much more of the component model has been implemented and is now fuzzed.
+
+This release also has a number of improvements to Cranelift compile times. With
+the [sightglass benchmark suite][sightglass] benchmarks show a 30-40%
```
Comment (cfallin) [inline RELEASES.md]: s/sightglass/Sightglass/

**Comment by alexcrichton** [comment]
In talking with @fitzgen I dropped the perf-related notes here. Building the `bench-api` crate with `--no-default-features`, notably disabling the `shuffling-allocator`, shows that `bz2` and `pulldown-cmark` had no changes in compile time and `spidermonkey`/`intgemm-simd` had very small changes (<5%), so not worth calling out in the relnotes

File: RELEASES.md [hunk 2]
```diff
@@ -14,9 +14,54 @@ Unreleased.
...
 Unreleased.
 
+This was a relatively quiet release in terms of user-facing features where most
+of the work was around the internals of Wasmtime and Cranelift. Improvements
+internally have been made along the lines of:
+
+* Many more instructions are now implemented with ISLE instead of handwritten
+  lowerings.
+* Many improvements to the cranelift-based fuzzing.
+* Many platform improvements for s390x for running `rustc_codegen_cranelift`
```
Comment (cfallin) [inline RELEASES.md:24]: call out SIMD for s390x in the top-notes here?

**Review by cfallin — APPROVED** [review]

File: RELEASES.md (hunk 1 above)
Comment (cfallin) [reply RELEASES.md]: Ah this is outdated, resolving

**Pull request merged**